### DISCUSSION TITLE
Build failed, Encountered error while archiving for device.

### DISCUSSION THREAD

**Discussion by kobimeridor** [codemagic-ci-cd/codemagic-docs#1543] [Q&A] [answered]
Build command -
`flutter build ipa --release \

  --build-name=8.0.0 \

  --build-number=$(($(app-store-connect get-latest-testflight-build-number "$APP_ID" --pre-release-version="5.0.0") + 1)) \

  --export-options-plist=/Users/builder/export_options.plist`

fail note- 
`Encountered error while archiving for device.`

In the log I have the following warning- , 
`The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0,` 
But in all my files IPHONEOS_DEPLOYMENT_TARGET is set to 11.0

Some builds before I had the error- 
`Running "flutter pub get" in flutter_tools...                       3.1s`
`Resolving dependencies...`
`The current Dart SDK version is 2.18.6.`  
But in my file it was set to 2.19 and I changed to
 `sdk: ">=2.18.0 <3.0.0" and didn't have this error

Its looking like the build base on setup that is not defined in my files`
Thanks
@maks

**Comment by maks** [comment]
Thanks for moving the discussion here.

The error is occurring because you are using an older Flutter version and that version of the Flutter SDK uses version `2.18.6` of Dart, so it can't meet your pubspec's requirement of 2.19. The latest stable version of Flutter (`3.7.9`) includes Dart version `2.19.6` so if you specify that in your Codemagic workflow you should no longer get the above error.

The way to set the version of the Flutter SDK that your workflow uses is either in the GUI if you are using the Workflow Editor or in your YAML within the `environment` property, eg.
```yaml
  environment:
    flutter: stable
    ...
```

**Comment by kobimeridor** [comment, marked as answer]
Thanks
This is environment section in Codemagic.yaml
What should I fix?

`    environment:
      groups:
        - A
      vars:
        appP: $APP_STORE_CONNECT_PRIVATE_KEY
        appK: $APP_STORE_CONNECT_KEY_IDENTIFIER
        appI: $APP_STORE_CONNECT_ISSUER_ID
      ios_signing:
        distribution_type: app_store # development # ad_hoc # or: ad_hoc | development | enterprise app_store
        bundle_identifier: com.allergy # QD2YB6G8K8.com.allergy # com.allergy
      flutter: 3.3.10
      xcode: 14.2
      cocoapods: 1.11.3
`

**Reply by maks** [reply]
Thanks for posting that section of your `codemagic.yaml`.

The property I was referring to is the `flutter:` one, Currently you have it set to a value of `3.3.10` which is a version which I mentioned has an older version of Dart. If you change that to `stable` it will use the current latest version in the stable channel, which as of today is `3.7.9`.

BUT please be aware that if you have code or package dependencies that require for some reason the `3.3.10` version of Flutter it may not work with `3.7.9`. Since I am not aware of  your app's code that's something that you need to check yourself.

**Reply by kobimeridor** [reply]
Thanks 
I updated to `stable` , started a new build and will update you

**Reply by kobimeridor** [reply]
Sorry, The build failed again
`Encountered error while archiving for device.`
The other warnings looking the same as before

link to Build- 
https://codemagic.io/app/63fe1398afbcfe7901b9478f/build/642d4a2c6e20b15303755400

**Comment by kobimeridor** [comment]
Is it Posible the `export-options-plist` in the build  makes a problem?

`             flutter build ipa --release \

            --build-name=8.0.0 \

            --build-number=$(($(app-store-connect get-latest-testflight-build-number "$APP_ID" --pre-release-version="5.0.0") + 1)) \

            --export-options-plist=/Users/builder/export_options.plist`


@maks

**Comment by kobimeridor** [comment]
Is it posible the this is the problem?
`    /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-BC363FF324E52E62005DC28B.sh: line 2: ./getsocial-sdk7.sh: No such file or directory
    Command PhaseScriptExecution failed with a nonzero exit code`
    @maks

**Reply by maks** [reply]
Oh! that's your post I just answered on SO.
Yes that will be an issue. Per my SO answer, the error message is saying that the file is not at the relative path of ` ./getsocial-sdk7.sh`
In your SO question, I note that the error message says the path of ` ios/getsocial-sdk7.sh` is not found so I guess you changed it after posting here. I would check exactly what the path is to that shell script relative to the top level directory in your Flutter apps git repo.

**Comment by kobimeridor** [comment]
Hi,
The following path has an error 

`    /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-BC363FF324E52E62005DC28B.sh: line 2: /Users/builder/clone/ios/getsocial-sdk7.sh: No such file or directory`

Is it possible that the path is ok and the file is not there?
How can I copy the file to this directory? 
@maks

**Comment by kobimeridor** [comment]
Hello
I added the script getsocial-sdk7.sh file to IOS directory and looks like the file was found.
Now i have a new error
`    /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-BC363FF324E52E62005DC28B.sh: line 2: /Users/builder/clone/ios/getsocial-sdk7.sh: Permission denied`

Where is permission definitions?
@maks

**Reply by maks** [reply]
This is standard Unix permissions.

You need to mark the file as executable, on the command line this is `chmod +x filename` and then commit the change in git.

**Comment by kobimeridor** [comment]
Thanks
I'm using windows
Where I need to write the command?
And what command do I use?

**Reply by maks** [reply]
On windows you should be able to do that using got itself if you are using a fairly up to date version of git, please see: https://stackoverflow.com/a/38285462/85472

**Comment by kobimeridor** [comment]
Thanks 
I will I try and update

**Comment by kobimeridor** [comment]
Thanks! 
I passed this issue
Kobi
@maks
